q2_types/feature_data/_format.py

```diff
@@ -131,23 +134,44 @@ def sniff(self):
 
 
 class DNAFASTAFormat(model.TextFileFormat):
-    def sniff(self):
-        filepath = str(self)
-        sniffer = skbio.io.io_registry.get_sniffer('fasta')
-        if sniffer(filepath)[0]:
-            generator = skbio.io.read(filepath, constructor=skbio.DNA,
-                                      format='fasta', verify=False)
-            try:
-                for seq, _ in zip(generator, range(5)):
-                    pass
-                return True
-            # ValueError raised by skbio if there are invalid DNA chars.
-            except ValueError:
-                pass
+    def _validate_lines(self, max_lines):
+        FASTADNAValidator = re.compile(r'[ACGTURYKMSWBDHVN\-\.]+\n?')
+        last_line_was_ID = False
 
-        # Empty files are ok also
-        empty_sniffer = skbio.io.io_registry.get_sniffer('<emptyfile>')
-        return empty_sniffer(filepath)[0]
+        with open(str(self), 'rb') as fh:
+            try:
+                first = fh.read(2)
+                first = first.decode('utf-8')
+                # Empty files should validate
+                if first == '\n' or first == '':
+                    return
+                if first[0] != '>':
```

Comment:
I think this is a really good approach, however there is something that is going to mess up the logic which plagued us for a while. The [utf8 byte order mark](https://en.wikipedia.org/wiki/Byte_order_mark#UTF-8) which wastes 3 bytes to tell us that there is no byte order.

I think you can work around this however: 
  1. Read 6 bytes, so that you read through this hypothetical BOM. Then determine if the first three bytes are a BOM, ignore them if that's the case.
  2. Instead of checking against a specific newline, calculate the length we expect (minus BOM), if its shorter than whatever we expected, then it must be an empty file irrespective of newline style. So with BOM we get less than 6 bytes (two bytes of windows newline) and without BOM we get less than 3 bytes (allowing two bytes for the windows newline). This has the added advantage of avoiding decoding anything as we're checking the first few bytes, which means we won't accidentally clip a longer code-point and misidentify the problem. They are doomed regardless, but the issue is that the first character isn't `>`, not that its not UTF-8 (which it may still be).
 3. Compare the first "real" byte against the ascii/unicode byte spelled `b'>'` instead of `'>'`, that should work and now you can avoid decoding the first few bytes (which runs into the BOM problem and potentially cutting through a character).

4. Finally, use `utf-8-sig` as your encoding elsewhere, this silently ignores the BOM.